lib/msal-angular/src/msal.navigation.client.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { NavigationClient, NavigationOptions, UrlString } from "@azure/msal-browser";
+import { Router } from "@angular/router";
+
+/**
+ * Custom navigation used for Angular client-side navigation.
+ * See performance doc for details:
+ * https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/lib/msal-angular/docs/v2-docs/performance.md
+ */
+export class MsalCustomNavigationClient extends NavigationClient {
+    private router: Router;
+
+    constructor(router: Router) {
+        super();
+        this.router = router;
+    }
+
+    async navigateInternal(url:string, options: NavigationOptions): Promise<boolean> {
```

Comment:
Do we need these `NavigationOptions` here? I know they're part of msal-browser's `NavigationClient`, but they are failing the linter here because they're not being used.

Reply:
I don't think you should need to provide it if its not used? Does typescript complain if you leave it out?

Reply:
You should use at least the history flag

Reply:
Noted. `navigateUrl` now uses `noHistory` flag.